### DISCUSSION TITLE
Is possible to add plugins with additional dependencies?

### DISCUSSION THREAD

**Discussion by juangon** [Kong/kong-operator#841] [Q&A] [answered]
From 1.4.x there is a new possibility for creating custom plugins by just adding them to a simple Dockerfile and create some Custom Resources (https://docs.konghq.com/gateway-operator/1.4.x/guides/plugin-distribution/).

My question is, if there are some additional dependencies (maybe some "resty.X") does this method still apply or we would need to create a custom kong-gateway image with that plugin as stated here? https://docs.konghq.com/gateway/latest/plugin-development/distribution/.

Is there a list with available dependencies when developing custom plugins that we can use out of the box? 

Thanks very much!

**Comment by programmer04** [comment, marked as answer]
Currently `KongPluginInstallation` is a mechanism for distributing code only. Dependency has to be installed on an image to make it available for plugins to use. The easiest way it's to do it with `luarocks` which is available on a distributed Kong Gateway image. Presently automated installation of dependency is not available.

> If the above answers your question please click Mark as answer button to increase visibility for others.

**Comment by juangon** [comment]
Thanks for the answer
